### PR TITLE
`attestation` configuration format change for server type diversity

### PR DESCRIPTION
**Pull request by jumaffre**
Follow-up from #4277 

This PR changes the the format of the `cchost` JSON configuration new `snp_endorsements_servers` section so that not all servers have to be of the same type, which was the case previously. Operators can now specify an Azure server first, followed by an AMD server as fallback. 

As this is now a list of `{"type": "...", "url": "..."}` objects, these can be specified as `--snp-endorsements-servers type:url` to the `sandbox.sh` script.

### CODE REVIEW

**Comment by achamayou** [comment]
Seems like something that would benefit from a changelog entry.

**Review by achamayou — APPROVED** [review]

File: tests/reconfiguration.py [hunk 1]
```diff
@@ -218,20 +218,18 @@ def test_add_node_amd_endorsements_endpoint(network, args):
 
     args_copy = deepcopy(args)
     test_vectors = [
-        ("AMD", ["kdsintf.amd.com"], True),
-        ("AMD", ["invalid.amd.com"], False),
-        ("AMD", ["invalid.amd.com", "kdsintf.amd.com"], True),  # Fallback server
+        (["AMD:kdsintf.amd.com"], True),
```
[the remark this answers is not in the document]
Comment (jumaffre) [reply tests/reconfiguration.py:221]: In the Azure case, there's multiple endpoints that can be queried, at different geographical locations and depending on whether this is a test or production instance, hence this here.

[the remark this answers is not in the document]
Comment (achamayou) [reply tests/reconfiguration.py:221]: It's also nice to be able to fallback if one or more doesn't work at first!

**Comment by jumaffre** [comment]
> Seems like something that would benefit from a changelog entry.

Because this went into `main` yesterday, I didn't add a specific entry to the changelog to correct this. I'll add a link to this PR in the existing entry though.

**Pull request merged**